pkg/storage/migrate/migrate.go

```diff
@@ -0,0 +1,154 @@
+package migrate
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/url"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/go-sql-driver/mysql"
+	"github.com/pressly/goose/v3"
+
+	"github.com/openfga/openfga/assets"
+	"github.com/openfga/openfga/pkg/storage/sqlite"
+)
+
+type MigrationConfig struct {
+	Engine        string
+	URI           string
+	TargetVersion uint
+	Timeout       time.Duration
+	Verbose       bool
+	Username      string
+	Password      string
+}
+
+// RunMigration runs the migrations for the given config. This function is exposed to allow embedding openFGA
+// into applications and manage OpenFGA's database schema migrations directly. When OpenFGA is used as a library,
+// the embedding application may have its own migration system that differs from OpenFGA's use of goose.
+// By exposing this function, applications can:
+// 1. Explicitly control when OpenFGA migrations run
+// 2. Integrate OpenFGA's schema updates into their own migration workflows
+// 3. Perform versioned upgrades of the schema as needed
+// The function handles migrations for multiple database engines (postgres, mysql, sqlite) and supports
+// both upgrading and downgrading to specific versions.
+func RunMigration(cfg MigrationConfig) error {
```

Comment:
```suggestion
func RunMigrations(cfg MigrationConfig) error {
```

Reply:
Are there tests within in cmd/migrate that make sense to move closer to this?

Reply:
I ended up moving `TestMigrateCommandRollbacks` since it was testing the dbs. The others tests are testing viper and command for issuing the config to parse it.